indico/modules/categories/controllers/management.py

```diff
@@ -247,3 +264,83 @@ def _process(self):
+              .format(len(self.events)), 'success')
+        return jsonify_data(flash=False, redirect=url_for('.manage_content', self.category))
+
+
+class RHSplitCategory(RHManageCategoryBase):
+    def _checkParams(self):
+        RHManageCategoryBase._checkParams(self)
+        self.category_events = {x.id: x for x in self.category.events}
+        if request.form.get('all_selected') == '1':
+            self.event_ids = self.category_events.viewkeys()
+        else:
+            self.event_ids = set(map(int, request.form.getlist('event_id')))
+
+    def _process(self):
+        all_selected = not bool(self.category_events.viewkeys() - self.event_ids)
+        form = SplitCategoryForm(move_all=all_selected, event_id=self.event_ids)
+        if form.validate_on_submit():
+            selected_events = self.category_events.viewkeys() & self.event_ids
+            not_selected_events = self.category_events.viewkeys() - self.event_ids
+            if selected_events:
+                self._move_events(selected_events, form.first_category.data)
+            if not_selected_events:
+                self._move_events(not_selected_events, form.second_category.data)
+            if all_selected:
+                flash(_('Your events have been moved successfully to category {}').format(form.first_category.data),
+                      'success')
+            else:
+                flash(_('Your events have been split between {} and {}').format(form.first_category.data,
+                                                                                form.second_category.data), 'success')
+            return jsonify_data(flash=False, redirect=url_for('.manage_content', self.category))
+        return jsonify_form(form)
+
+    def _move_events(self, event_ids, category_title):
+        category = Category(title=category_title, parent_id=self.category.id)
+        for event_id in event_ids:
+            event = self.category_events[event_id]
+            event.move(category)
+        db.session.flush()
+
+
+class RHMoveEvents(RHManageCategoryBase):
+    def _checkParams(self):
+        RHManageCategoryBase._checkParams(self)
+        self.category_events = {x.id: x for x in self.category.events}
+        if request.json.get('all_selected') == 1:
+            self.event_ids = self.category_events.viewkeys()
+        else:
+            self.event_ids = set(map(int, request.json.get('event_id')))
+
+    def _process(self):
+        new_category = Category.find(id=request.json.get('category_id')).first_or_404()
+        count = 0
+        for event_id in self.event_ids:
+            if event_id in self.category_events:
+                self.category_events[event_id].move(new_category)
+                count += 1
+        flash(ngettext('You have moved one event to {} category',
+                       'You have moved {} events to {} category', count).format(count,
```

Comment:
always use positional or better named placeholders in i18n strings that contain more than one placeholder. otherwise it's impossible to translate it properly in some cases